src/orchard/inspect.clj

```diff
@@ -575,11 +575,13 @@
 (defn- shorten-member-string [member-string, ^Class class]
   ;; Ugly as hell, but easier than reimplementing all custom printing that
   ;; java.lang.reflect does.
-  (-> member-string
-      (string/replace #"[\w\.]+\.(\w+\()" "$1") ;; remove class from method name
-      (string/replace #"[\w\.]+\.(\w+)$" "$1") ;; remove class from field name
-      (string/replace (.getCanonicalName class) (.getSimpleName class))
-      (string/replace #"java.lang.([A-Z])" "$1")))
+  (as-> member-string s
+    (string/replace s #"[\w\.]+\.(\w+\()" "$1") ;; remove class from method name
+    (string/replace s #"[\w\.]+\.(\w+)$" "$1") ;; remove class from field name
+    (if-let [canonical (.getCanonicalName class)]
```

Comment:
Maybe add a comment here explaining when we won't have the canonical name?

Reply:
Left a comment, the list is in the javadoc for `.getCanonicalName`.